cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -1470,12 +1470,16 @@
 (rule (lower (icmp (IntCC.SignedLessThan) a @ (value_type (vec128 ty)) b))
       (x64_pcmpgt ty b a))
 (rule (lower (icmp (IntCC.UnsignedGreaterThan) a @ (value_type (vec128 ty)) b))
-      (let ((max Xmm (x64_pmaxu ty a b))
+      ;; N.B.: we must manually prevent load coalescing of these operands; the
+      ;; register allocator gets confused otherwise. TODO:
+      ;; https://github.com/bytecodealliance/wasmtime/issues/3953.
+      (let ((max Xmm (x64_pmaxu ty (put_in_xmm a) (put_in_xmm b)))
             (eq Xmm (x64_pcmpeq ty max b))
             (all_ones Xmm (vector_all_ones ty)))
            (x64_pxor eq all_ones)))
 (rule (lower (icmp (IntCC.UnsignedLessThan) a @ (value_type (vec128 ty)) b))
-      (let ((min Xmm (x64_pminu ty a b))
+      ;; N.B.: see note above.
+      (let ((min Xmm (x64_pminu ty (put_in_xmm a) (put_in_xmm b)))
```

Comment:
Likewise here, I'd bind the registers as separate locals then use them here and in the pcmpeq below.